src/python/CMakeLists.txt

```diff
@@ -326,79 +326,103 @@ if(PYTHONINTERP_FOUND)
+          endif(MATPLOTLIB_FOUND)
+        else(PYDATA_SPHINX_THEME_FOUND)
+          message("++ Python documentation module will not be compiled because pydata_sphinx_theme was not found")
           set(GUDHI_MISSING_MODULES ${GUDHI_MISSING_MODULES} "python-documentation" CACHE INTERNAL "GUDHI_MISSING_MODULES")
-        endif(MATPLOTLIB_FOUND)
-      else(PYDATA_SPHINX_THEME_FOUND)
-        message("++ Python documentation module will not be compiled because pydata_sphinx_theme was not found")
+        endif(PYDATA_SPHINX_THEME_FOUND)
+      else(SPHINX_PARAMLINKS_FOUND)
+        message("++ Python documentation module will not be compiled because sphinxcontrib-paramlinks was not found")
         set(GUDHI_MISSING_MODULES ${GUDHI_MISSING_MODULES} "python-documentation" CACHE INTERNAL "GUDHI_MISSING_MODULES")
-      endif(PYDATA_SPHINX_THEME_FOUND)
-    else(SPHINX_PARAMLINKS_FOUND)
-      message("++ Python documentation module will not be compiled because sphinxcontrib-paramlinks was not found")
+      endif(SPHINX_PARAMLINKS_FOUND)
+    else(SPHINX_PATH)
+      message("++ Python documentation module will not be compiled because sphinx and sphinxcontrib-bibtex were not found")
       set(GUDHI_MISSING_MODULES ${GUDHI_MISSING_MODULES} "python-documentation" CACHE INTERNAL "GUDHI_MISSING_MODULES")
-    endif(SPHINX_PARAMLINKS_FOUND)
-  else(SPHINX_PATH)
-    message("++ Python documentation module will not be compiled because sphinx and sphinxcontrib-bibtex were not found")
-    set(GUDHI_MISSING_MODULES ${GUDHI_MISSING_MODULES} "python-documentation" CACHE INTERNAL "GUDHI_MISSING_MODULES")
-  endif(SPHINX_PATH)
+    endif(SPHINX_PATH)
 
+    # Cubical
+    add_test(NAME periodic_cubical_complex_barcode_persistence_from_perseus_file_example_py_test
+             WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+             COMMAND ${CMAKE_COMMAND} -E env "${GUDHI_PYTHON_PATH_ENV}"
+             ${PYTHON_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/example/periodic_cubical_complex_barcode_persistence_from_perseus_file_example.py"
+             --no-barcode -f ${CMAKE_SOURCE_DIR}/data/bitmap/CubicalTwoSphere.txt)
+
+    add_test(NAME random_cubical_complex_persistence_example_py_test
+             WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+             COMMAND ${CMAKE_COMMAND} -E env "${GUDHI_PYTHON_PATH_ENV}"
+             ${PYTHON_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/example/random_cubical_complex_persistence_example.py"
+             10 10 10)
+
+    add_gudhi_py_test(test_cubical_complex)
+
+    # Datasets are fetched for these tests 
+    if(SKLEARN_FOUND AND WITH_GUDHI_REMOTE_TEST)
+      add_gudhi_py_test(test_sklearn_cubical_persistence)
+
+      add_test(NAME cubical_complex_sklearn_itf_py_test
+               WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+               COMMAND ${CMAKE_COMMAND} -E env "${GUDHI_PYTHON_PATH_ENV}"
+               ${PYTHON_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/example/cubical_complex_sklearn_itf.py")
```

Comment:
Is it the same code that was in the doc? If so, isn't it a bit too long for a test? Should it include an assertion at the end that the classification score was not too horrible?

Reply:
Yes, I moved the code from the doc in an example file that is now included in the doc and tested by the test suite only if scikit-learn and `WITH_GUDHI_REMOTE_TEST` (as data are fetched).

The test is quite long (90 sec. on my laptop, 490 sec. on circleCI, 373 sec. on Azure Windows, 230 sec. on azure OSx), but it worth it because this example changed a lot since its introduction in 3.6.0 and has been broken several times. I don't want to add it to sphinx_py_test that already takes a lot of time.

Don't you think this is a good compromise ?

Reply:
For someone running the testsuite in parallel, this doubles the time the whole testsuite takes, that's too long.
Now if you don't use all the images but only `X=X[:10000]` (same for `y`, and it doesn't have to be exactly 10000 I picked that randomly), the running time becomes much more acceptable, and the score is still good.